### ISSUE TITLE
[Feature Request] Symmetric option for Linear and Polar patterns

### ISSUE THREAD

**Issue by kadet1090** [FreeCAD/FreeCAD#10442]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Full version info

```shell
OS: Arch Linux (awesome)
Word size of FreeCAD: 64-bit
Version: 0.21.0.33675 (Git)
Build type: Release
Branch: makepkg
Hash: 41b058e2087de60dc8fef3d3e68c7d0129e13abf
Python 3.11.3, Qt 5.15.10, Coin 4.0.1, Vtk 9.2.6, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * Assembly3 0.12.0
  * IconThemes
  * Glass (Disabled)
  * lattice2 1.0.0
  * A2plus 0.4.60n
```

### Subproject(s) affected?

PartDesign

### Problem description

For some patterns it'd be nice to be able make them symmetrical to the original feature(s). I think that the best option would be to implement it as additional boolean property / checkbox for the patterns - this way we could achieve similar UX between two possible pattern kinds. 

#### How it should behave?
There are few options how such "Symmetric" option could work, for example, when we have:
```
- Angle set to 90
- Mode set to Overall Angle
- Occurrences set to 2
- Symmetric option on
```

Should we have:
a) 2 features, 90 degrees apart? We would have to "move" original feature.
b) 3 features spanning across 90 degrees?
c) 3 features spanning across 180 degrees - 2 per symmetry side, each side spans on 90 degrees? Then we'd have to constraint maximum angle to 180, not mentioning that angle would no longer be overall.

![image](https://github.com/FreeCAD/FreeCAD/assets/747404/2fd8bc1e-5e14-4e5c-b9f8-f983519ac8dd)

Personally I think that a) seems to be most useful, c) also could be an option but it could be easily done now using multi transform with pattern + mirror. I however think that having that option as simple checkbox would be great for product usability. 

If we choose option a) for even number of occurrences, I see few potential problems:

#### Relation to Topological Naming
As there will no longer be feature in original place, instead pattern would be distributed evenly across length or angle. Will it have impact on topological naming, as it'd be hard to name any created feature as original?

#### Non-Patterned feature
We can create pattern only for some features - how about case where we have some dependency on the patterned feature which will not be a part of pattern? Should we even care, or consider it as error from the users side which should be corrected (feature should be created after pattern).

![image](https://github.com/FreeCAD/FreeCAD/assets/747404/46de98f8-9c12-4fe1-9cd8-3a1e8ed18843)

For this example the pocketed circle will move so peg will no longer be placed correctly.

### Anything else?

Forum discussion: https://forum.freecad.org/viewtopic.php?t=68621&start=13

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mbello** [comment]
I always missed this feature. But what if, instead of "Symmetric", which has all the corner cases you mentioned, we had a simpler "Mirrored" boolean?

**Comment by grubertechnic** [comment]
There is already a mirror feature for patterns. But after rethinking the cases, a symmetric makes sense, especially when more options of transformation will be added. But it makes only sense IN the pattern itself.



> But what if, instead of "Symmetric", which has all the corner cases you mentioned, we had a simpler "Mirrored" boolean?

That is the point, we already have mirror, for all cases there will be needed a symmetric.

**Comment by kadet1090** [comment]
Mirrored would indeed be much simpler, but it's not edge-cases free - see case with polar pattern for angles above 180, especially 360 where it'd create doubled features. If we won't be able to converge on behavior for symmetric case I'd personally still like to see "Mirrored" checkbox. At least it'd be easier to create such patterns and require less operations from the user - which always is better from UX perspective.

**Comment by adrianinsaval** [comment]
I'm not convinced that we need or want this feature, we have multi transforms where you can make a mirror out of a pattern. This is duplicating functionality in a different place.

**Comment by obelisk79** [comment]
I agree, duplicating functionality is generally a bad idea and is even addressed in the design guide if I'm remembering right. Perhaps @kadet1090 can explain how this is different or perhaps is a superior approach?

**Comment by kadet1090** [comment]
Yes, implemented as Mirrored it is duplicating thing that we already functionally have, however making it easier to discover. Multi transform works but the workflow is not easy to find, not obvious - and it is still mentally two steps to define. 

That said, this issue was mostly about symmetry so options a and b which are hard to define correctly. I even stated in the description that c is already easy to achieve and thus not as useful as for example option A which allows to easily distribute feature knowing the symmetry and overall length which now must be done using expressions.

Honestly, I don't think that we agreed on how it should work here and that is why I never bothered to implement it.

**Comment by adrianinsaval** [comment]
I agree multi transform has bad UX, but the solution is to improve the UX there not to hack duplicated functionality somewhere else.

**Comment by adrianinsaval** [comment]
On the other hand Pad and revolve offer a symmetry option, so it might be justified after all.

**Comment by Danielsvgx** [comment]
I wish symmetry/centered/bilateral linear arrays were a thing too.

I am trying to create rows of hex pockets centered on a face.

The intuitive way to do it would be to create a sketch with my hole shape at the location where I want the center of my row located, then have a linear pattern of however many pockets I want stretched however far on either side of my center pocket and I can make that hole count even or odd without worrying about offsetting my sketch.

With linear array, I can create the holes to one side of my sketch but when it comes time to mirror that array, instead of mirroring relative to the original feature, it mirrors relative to other plane, way out into la-la-land, with none of the mirror planes offered in the transform tool being what I need to make it work that way.

If I wanted to use mirror, I would have to go through the extra steps of creating a datum plane that cuts my feature in half where I want the array mirrored. This only works if my feature shape is symmetrical across that plane and only works for odd numbers of features. If I wanted an even number of holes, I would have to go through extra steps of offsetting my starting sketch by half the spacing amount from my mirror plane. That is a lot of extra stuff to make parametric to work-around the lack of such a simple option.

Also, if my hexagon wasn't symmetrical with the mirror plane, mirroring would turn the center hexagon into a 12-gon with the mirrored pockets reversed. Not what I am looking for.

Another option is to create a second linear array with reversed direction, which requires manually lining up the first hole of the first array with the last hole of the second. Again, this only works for odd numbers of features. If I wanted an even number, I would have to offset my sketch by half the spacing and then manually place the linear copy at the correct distance from the first one.

Both are horrible from an end-user's standpoint: 3-10 extra steps depending on odd vs even and the chosen work-around method, 30+ minutes researching whether this is actually as-intended, an omission, an open/pending issue, etc., 30+ minutes looking for the most effective work-around and 60+ minutes figuring out how to use those work-arounds without breaking other stuff. More work-around steps will likely introduce more breakage to fix in the future too. All of those extra steps also make the history balloon up quite fast.

I have 10 different rows of such features to make. Would have taken a minute if linear arrays could be bilateral/centered, will take ~10 because they are not and odd/evens have to be handled differently unless I spent an hour making them parametric instead.
